src/nlsolve/type.jl

```diff
@@ -18,6 +18,8 @@ mutable struct NLSolverCache{rateType,uType,W,uToltype,cType,gType} <: AbstractN
   tmp::uType
   b::uType # can be aliased with `k` if no unit
   k::rateType
+  zs::Array
```

Comment:
this won't be inferred. Just parameterize it, and change the `[]`s to `nothing`

Reply:
On changing `Array` to `Array{uType}` and `[]` to `nothing`, I get this error:
```
ERROR: LoadError: MethodError: no method matching OrdinaryDiffEq.NLSolverCache(::Float64, ::Float64, ::Int64, ::Int64, ::Int64, ::Bool, ::Float64, ::Float64, ::Int64, ::Int64, ::Float64, ::Float64, ::Float64, ::Float64, ::Float64, ::Float64, ::Nothing, ::Nothing)
Closest candidates are:
  OrdinaryDiffEq.NLSolverCache(::uToltype, ::uToltype, ::Int64, ::Int64, ::Int64, ::Bool, ::uType, ::W, ::gType, ::cType, ::uToltype, ::uType, ::uType, ::uType, ::uType, ::rateType, !Matched::Array{uType,N} where N, !Matched::Array{uType,N} where N) where {rateType, uType, W, uToltype, cType, gType} at /home/kanav/.julia/dev/OrdinaryDiffEq/src/nlsolve/type.jl:4
```

Reply:
Use ‘zsType’, and put it in the above curly bracket.

Reply:
no, parameterize the field